Add a link to learn about pointers

illicitonion opened the pull request:
Contributes to #123

sre-is-laura commented:
i wonder if a basic primer on what stack/heap etc are might also help with this
this isn't bad: https://courses.engr.illinois.edu/cs225/fa2022/resources/stack-heap/

illicitonion commented:
> i wonder if a basic primer on what stack/heap etc are might also help with this this isn't bad: https://courses.engr.illinois.edu/cs225/fa2022/resources/stack-heap/

Good shout! Added, and added a little commentary about the differences between the C++ used in examples and Go they will be writing, with a few breadcrumb links to follow up if they're interested.
